yarn-project/bb-prover/src/avm_proving_tests/avm_contract_updates.test.ts

```diff
@@ -46,22 +42,17 @@ describe('AVM WitGen & Circuit - contract updates', () => {
     nextClassId: Fr,
     blockOfChange: number,
   ) => {
-    const sharedMutableSlot = await deriveStorageSlotInMap(new Fr(UPDATED_CLASS_IDS_SLOT), contractAddress);
+    const { sharedMutableSlot } = await SharedMutableValuesWithHash.getContractUpdateSlots(contractAddress);
 
     const valueChange = new ScheduledValueChange([previousClassId], [nextClassId], blockOfChange);
     const delayChange = ScheduledDelayChange.empty();
+    const sharedMutableValuesWithHash = new SharedMutableValuesWithHash(valueChange, delayChange);
+
     const writeToTree = async (storageSlot: Fr, value: Fr) => {
       await tester.setPublicStorage(ProtocolContractAddress.ContractInstanceDeployer, storageSlot, value);
     };
-    await valueChange.writeToTree(sharedMutableSlot, writeToTree);
-    await delayChange.writeToTree(sharedMutableSlot, writeToTree);
-
-    const updatePreimage = [delayChange.toField(), ...valueChange.toFields()];
-    const updateHash = await poseidon2Hash(updatePreimage);
-
-    const hashSlot = computeSharedMutableHashSlot(sharedMutableSlot, UPDATES_SCHEDULED_VALUE_CHANGE_LEN);
 
-    await writeToTree(hashSlot, updateHash);
+    await sharedMutableValuesWithHash.writeToTree(sharedMutableSlot, writeToTree);
```

Comment:
Writing the values and the hash separately was too ugly so I decided to crated the SharedMutableValuesWithHash class.